src/main/java/com/internet/shop/controller/order/CompleteOrderController.java

```diff
@@ -0,0 +1,31 @@
+package com.internet.shop.controller.order;
+
+import com.internet.shop.lib.Injector;
+import com.internet.shop.model.Order;
+import com.internet.shop.model.ShoppingCart;
+import com.internet.shop.service.interfaces.OrderService;
+import com.internet.shop.service.interfaces.ShoppingCartService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class CompleteOrderController extends HttpServlet {
+    private static final Long USER_ID = 1L;
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private ShoppingCartService shoppingCartService
+            = (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
+    private OrderService orderService = (OrderService) injector.getInstance(OrderService.class);
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        Long cartId = Long.parseLong(req.getParameter("cartId"));
+        ShoppingCart shoppingCart = shoppingCartService.get(cartId);
+        Order order = orderService.completeOrder(shoppingCart);
+        order.setUserId(shoppingCart.getUserId());
+        shoppingCartService.clear(shoppingCart);
+        resp.sendRedirect(req.getContextPath() + "/user/orders");
```

Comment:
Don't you clear it at the service layer in the orderService.complete() method? 
Same about id setting